pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -42,6 +57,360 @@ type Adapter struct {
+		if localURL.Kind != envinfo.DOCKER {
+			deploymentPort = strconv.Itoa(localURL.Port)
+			break
+		}
+	}
+	return deploymentPort
+}
+
+func substitueYamlVariables(baseYaml []byte, yamlSubstitutions map[string]string) []byte {
+	// TODO: Provide a better way to do the substitution in the manifest file(s)
+	for key, value := range yamlSubstitutions {
+		if value != "" && bytes.Contains(baseYaml, []byte(key)) {
+			klog.V(3).Infof("Replacing %s with %s", key, value)
+			tempYaml := bytes.ReplaceAll(baseYaml, []byte(key), []byte(value))
+			baseYaml = tempYaml
+		}
+	}
+	return baseYaml
+}
+
+func getNamedCondition(route *unstructured.Unstructured, conditionTypeValue string) map[string]interface{} {
+	status := route.UnstructuredContent()["status"].(map[string]interface{})
+	conditions := status["conditions"].([]interface{})
+	for i := range conditions {
+		c := conditions[i].(map[string]interface{})
+		klog.V(4).Infof("Condition returned\n%s\n", c)
+		if c["type"] == conditionTypeValue {
+			return c
+		}
+	}
+	return nil
+}
+
+// TODO: Create a function to wait for deploment completion of any unstructured object
+func (a Adapter) waitForManifestDeployCompletion(applicationName string, gvr schema.GroupVersionResource, conditionTypeValue string) (*unstructured.Unstructured, error) {
+	klog.V(4).Infof("Waiting for %s manifest deployment completion", applicationName)
+	w, err := a.Client.DynamicClient.Resource(gvr).Namespace(a.Client.Namespace).Watch(metav1.ListOptions{FieldSelector: "metadata.name=" + applicationName})
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to watch deployment")
+	}
+	defer w.Stop()
+	success := make(chan *unstructured.Unstructured)
+	failure := make(chan error)
+
+	go func() {
+		defer close(success)
+		defer close(failure)
+
+		for {
+			val, ok := <-w.ResultChan()
+			if !ok {
+				failure <- errors.New("watch channel was closed")
+				return
+			}
+			if watchObject, ok := val.Object.(*unstructured.Unstructured); ok {
+				// TODO: Add more details on what to check to see if object deployment is complete
+				// Currently only checks to see if status.conditions[] contains a condition with type = conditionTypeValue
+				condition := getNamedCondition(watchObject, conditionTypeValue)
+				if condition != nil {
+					if condition["status"] == "Fail" {
+						failure <- fmt.Errorf("manifest deployment %s failed", applicationName)
+					} else if condition["status"] == "True" {
+						success <- watchObject
+					}
+				}
+			}
+		}
+	}()
+
+	select {
+	case val := <-success:
+		return val, nil
+	case err := <-failure:
+		return nil, err
+	case <-time.After(30 * time.Second):
+		return nil, errors.Errorf("timeout while waiting for %s manifest deployment completion", applicationName)
+	}
+}
+
+// Build image for devfile project
+func (a Adapter) Deploy(parameters common.DeployParameters) (err error) {
+	namespace := a.Client.Namespace
+	applicationName := a.ComponentName + "-deploy"
```

Comment:
Need to agree on naming and document it

Reply:
In build with use the componentName directly as the name of the BuildConfig. Maybe we can leave `-deploy` for now, or even better check if a resource with that name exists and decide.